functors task (2)/student/include/impl.h

```diff
@@ -21,7 +21,7 @@ class DataBrowser
      */
     void userEnter(const std::string& userId)
     {  // awful, but in header for testing
-        m_dataReaders.emplace(userId, getSelector());
+        m_dataReaders.emplace( userId, std::unique_ptr<IDataSelector>( getSelector() ) );
```

Comment:
cheating, please don't modify this code

or you mean something is wrong in task interface?

Reply:
It was not compiling otherwise.

Reply:
Please provide compilation log and compiler version